File: python/day82.py
import re

user_input = input("Pleae enter your credit/debit card number: ")

clean_number = re.sub(r'/D', '', user_input)

if 13 <= len(clean_number) < 19:
  formatted_number = " ".join(clean_number[i:i+4] for i in range(0, len(clean_number), 4)) 
  print(f"Thank you! Saverd as: {clean_number}")
  print(f"Formatted View: {formatted_number}")
else:
  print("\nError: Invalid card length. Please enter a valid number.")

File: js/day82.js
const readline = require('readline');

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

rl.question("Please enter your credit/debit card number: ", (input) => {

  const cleanNumber = input.replace(/\D/g, '');

  if(cleanNumber.length >= 13 && cleanNumber.length <= 19){
    const formattedNumber = cleanNumber.replace(/(.{4})/g, '$1').trim();
    console.log(`Thank you! Saved as: ${cleanNumber}`);
    console.log(`Formatted View: ${formattedNumber}`);
  } else {
    console.log(`\nError: Invalid card length. Please enter a valid number.`);
  }
  
  rl.close();
});